src/Components/Blazor/WebAssembly.Authentication/src/AuthenticationManager.cs

```diff
@@ -0,0 +1,446 @@
+            {
+                State = AuthenticationState
+            });
+            switch (result.Status)
+            {
+                case RemoteAuthenticationStatus.Redirect:
+                    break;
+                case RemoteAuthenticationStatus.Success:
+                    Navigation.NavigateTo(returnUrl);
+                    break;
+                case RemoteAuthenticationStatus.Failure:
+                    var uri = Navigation.ToAbsoluteUri($"{ApplicationPaths.LoginFailedPath}?message={result.ErrorMessage}").ToString();
+                    Navigation.NavigateTo(uri);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        private async Task ProcessLoginCallback()
+        {
+            var url = Navigation.Uri;
+            var result = await AuthenticationService.CompleteSignInAsync(new RemoteAuthenticationContext<TAuthenticationState> { Url = url });
+            switch (result.Status)
+            {
+                case RemoteAuthenticationStatus.Redirect:
+                    // There should not be any redirects as the only time CompleteSignInAsync finishes
+                    // is when we are doing a redirect sign in flow.
+                    throw new InvalidOperationException("Should not redirect.");
+                case RemoteAuthenticationStatus.Success:
+                    await NavigateToReturnUrl(GetReturnUrl(result.State));
+                    break;
+                case RemoteAuthenticationStatus.OperationCompleted:
+                    break;
+                case RemoteAuthenticationStatus.Failure:
+                    var uri = Navigation.ToAbsoluteUri($"{ApplicationPaths.LoginFailedPath}?message={result.ErrorMessage}").ToString();
+                    Navigation.NavigateTo(uri);
+                    break;
+                default:
+                    throw new InvalidOperationException($"Invalid authentication result status '{result.Status}'.");
+            }
+        }
+
+        private async Task ProcessLogout(string returnUrl)
+        {
+            AuthenticationState.ReturnUrl = returnUrl;
+            var user = await AuthenticationService.GetCurrentUser();
+            var isauthenticated = user.Identity.IsAuthenticated;
+
+            if (isauthenticated)
+            {
+                var result = await AuthenticationService.SignOutAsync(new RemoteAuthenticationContext<TAuthenticationState> { State = AuthenticationState });
+                switch (result.Status)
+                {
+                    case RemoteAuthenticationStatus.Redirect:
+                        break;
+                    case RemoteAuthenticationStatus.Success:
+                        await NavigateToReturnUrl(returnUrl);
+                        break;
+                    case RemoteAuthenticationStatus.OperationCompleted:
+                        break;
```

Comment:
What does this scenario represent? If the operation completed, how is it not a success or a failure, and why is it OK just to do nothing?